trussworks/react-uswds
# feat: SiteAlert child enhancement

# Summary

Modifies the `SiteAlert` component to also accept simple text strings and multiple components as children. If the child is a simple string, the component will wrap it in `<p className="usa-alert__text">`, which aligns with what the [USWDS component](https://designsystem.digital.gov/components/site-alert/) is doing. (Single child components and multiple child components are deliberately left alone, as users may already be passing in content for the alert wrapped in a `p` tag.)

There is a desire to further enhance this functionality by creating an `AlertText` sub-component, but as I'm still trying to figure out how that could practically work (as it's possible to also use it in the `Alert` component as well but that might introduce a breaking change?), I'm making this intermediate PR since it's already a value add at this point.

## Related Issues or PRs

Addresses (but does not fully close, see above + issue discussion) #1191

## How To Test

Check out this branch, run `yarn test` to execute tests, and `yarn storybook` to see the updated component functionality in the additional stories for the `SiteAlert` component.
